### PR TITLE
MC-20696: add docs for probes

### PR DESCRIPTION
**Pull request by HassanHaidar**
#### Changes made
Add API docs for liveness and readiness probes

[MC-20696]: https://cloudops.atlassian.net/browse/MC-20696?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Pearl1594** [comment]
Do we still need to update the slate docs?

**Comment by HassanHaidar** [comment]
> Do we still need to update the slate docs?

Yes we still need to keep them up-to-date for a while until we officially deprecate them

**Review by lamminade — APPROVED** [review]

**Review by Pearl1594 — APPROVED** [review]

File: source/includes/stackpath/_workloads.md [hunk 1]
```diff
@@ -160,6 +173,32 @@ Attributes | &nbsp;
...
+`readinessProbe.periodSeconds`<br/>*integer* | The frequency of the probe in seconds.
+`readinessProbe.successThreshold`<br/>*integer* | The minimum consecutive successes required before a probe is considered successful after a failure. This must be 1 for liveness probes.
+`readinessProbe.failureThreshold`<br/>*integer* | The amount of failures seen before the probe is considered a failure.
+`readinessProbe.httpGet`<br/>*Object* | HTTP GET request executed against an endpoint running on an instance.
+`readinessProbe.httpGet.path`<br/>*string* | The path portion of the URL to request.
+`readinessProbe.httpGet.port`<br/>*integer* | The TCP port to query in the HTTP request.
+`readinessProbe.httpGet.httpHeaders`<br/>*List* | A list of HTTPs headers.
+`readinessProbe.httpGet.scheme`<br/>*enum* | HTTP scheme to use in the HTTP request. It could be either "HTTP" or "HTTPS".
+`readinessProbe.tcpSocket`<br/>*Object* | The instance's TCP port to connect to.
+`readinessProbe.tcpSocket.port`<br/>*Object* | The TCP port to query.
```
Comment (fyrnaga) [inline source/includes/stackpath/_workloads.md:201]: Should there be a mention that it is either httpGet or tcpSocket?

File: source/includes/stackpath/_workloads.md [hunk 2]
```diff
@@ -160,6 +173,32 @@ Attributes | &nbsp;
...
+`livenessProbe.periodSeconds`<br/>*integer* | The frequency of the probe in seconds.
+`livenessProbe.successThreshold`<br/>*integer* | The minimum consecutive successes required before a probe is considered successful after a failure. This must be 1 for liveness probes.
+`livenessProbe.failureThreshold`<br/>*integer* | The amount of failures seen before the probe is considered a failure.
+`livenessProbe.httpGet`<br/>*Object* | HTTP GET request executed against an endpoint running on an instance.
+`livenessProbe.httpGet.path`<br/>*string* | The path portion of the URL to request.
+`livenessProbe.httpGet.port`<br/>*integer* | The TCP port to query in the HTTP request.
+`livenessProbe.httpGet.httpHeaders`<br/>*List* | A list of HTTPs headers.
+`livenessProbe.httpGet.scheme`<br/>*enum* | HTTP scheme to use in the HTTP request. It could be either "HTTP" or "HTTPS".
+`livenessProbe.tcpSocket`<br/>*Object* | The instance's TCP port to connect to.
+`livenessProbe.tcpSocket.port`<br/>*Object* | The TCP port to query.
```
Comment (fyrnaga) [inline source/includes/stackpath/_workloads.md:188]: same here

File: source/includes/stackpath/_workloads.md (hunk 1 above)
Comment (fyrnaga) [reply source/includes/stackpath/_workloads.md:201]: at least for the create/update

Comment (HassanHaidar) [reply source/includes/stackpath/_workloads.md:201]: i'll add it

Comment (HassanHaidar) [reply source/includes/stackpath/_workloads.md:201]: d96ae48b3812b4333cdf73f2d55eeb98916e277d

File: source/includes/stackpath/_workloads.md (hunk 2 above)
Comment (HassanHaidar) [reply source/includes/stackpath/_workloads.md:188]: d96ae48b3812b4333cdf73f2d55eeb98916e277d

**Review by fyrnaga — APPROVED** [review]

**Pull request merged**